plugins/inputs/lvm/lvm.go

```diff
@@ -0,0 +1,287 @@
+package lvm
+
+import (
+	"encoding/json"
+	"fmt"
+	"os/exec"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+var (
+	execCommand = exec.Command
+)
+
+var sampleConfig = `
+## Use sudo to run LVM commands
+use_sudo = false
+`
+
+type LVM struct {
+	UseSudo bool
+}
+
+func (lvm *LVM) Description() string {
+	return "Read metrics about LVM physical volumes, volume groups, logical volumes."
+}
+
+func (lvm *LVM) SampleConfig() string {
+	return sampleConfig
+}
+
+func (lvm *LVM) Init() error {
+	return nil
+}
+
+func (lvm *LVM) Gather(acc telegraf.Accumulator) error {
+	if err := lvm.gatherPhysicalVolumes(acc); err != nil {
+		return err
+	} else if err := lvm.gatherVolumeGroups(acc); err != nil {
+		return err
+	} else if err := lvm.gatherLogicalVolumes(acc); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (lvm *LVM) gatherPhysicalVolumes(acc telegraf.Accumulator) error {
+	pvsCmd := "/usr/sbin/pvs"
+	args := []string{
+		"--reportformat", "json", "--units", "b", "--nosuffix",
+		"-o", "pv_name,vg_name,pv_size,pv_free,pv_used",
+	}
+	out, err := lvm.runCmd(pvsCmd, args)
+	if err != nil {
+		return err
+	}
+
+	var report pvsReport
+	err = json.Unmarshal(out, &report)
+	if err != nil {
+		return fmt.Errorf("failed to unmarshal physical volume JSON: %s", err)
+	}
+
+	for _, pv := range report.Report[0].Pv {
```

Comment:
Should check the length of `report.Report[0]` before trying to access the first element. Don't forget to update the same loops below.

Reply:
Ah right thank you!